client/me/purchases/pre-cancelation-dialog/index.tsx

```diff
@@ -0,0 +1,180 @@
+import { Dialog, Gridicon } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import getPlanFeatures from './get-plan-features';
+import './style.scss';
+
+interface PreCancelationDialogProps {
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	site: SiteExcerptData;
+	hasDomain: boolean;
+	isRefundable: boolean;
+	isAutoRenewing: boolean;
+	primaryDomain: string;
+	wpcomURL: string;
+}
+
+export const PreCancelationDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+	hasDomain,
+	isRefundable,
+	isAutoRenewing,
+	primaryDomain,
+	wpcomURL,
+}: PreCancelationDialogProps ) => {
+	const translate = useTranslate();
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const subTitle = translate( 'Subtitle' );
+
+	/**
+	 * Click events, buttons tracking and action.
+	 */
+	const clickCancelPlan = () => {
+		removePlan();
+	};
+
+	const clickCloseDialog = () => {
+		closeDialog();
+	};
+
+	/**
+	 * Domain redirect copy
+	 */
+	const domainFeature =
+		hasDomain && primaryDomain && wpcomURL ? (
+			<p>Warning: domain traffic will be redirected</p>
+		) : null;
+
+	/**
+	 * Dialog buttons
+	 */
+	const buttons = [
+		{
+			action: 'cancel',
+			label: translate( 'Cancel my plan' ),
+			onClick: clickCancelPlan,
+		},
+		{
+			action: 'close',
+			label: translate( 'Keep my plan' ),
+			onClick: clickCloseDialog,
+			isPrimary: true,
+		},
+	];
+
+	/**
+	 * Return the list of features that the user will lose by canceling their plan.
+	 *
+	 * @returns Fragment
```

Comment:
This isn't precise since it can return a null. I don't really think we need to right a formal JSDoc for an inline function, but if we want we should make it `@returns {Fragment|null}` or `@returns {Fragment?}

Reply:
good catch, I removed the JSDoc  return part